pkg/controller/controller.go

```diff
@@ -188,28 +226,30 @@ func GetTopController(ctx context.Context, dynamicClient dynamic.Interface, rest
 				return unstructuredObject, errors.New("this object could not be found for this object " + key)
 			}
 		}
-		return GetTopController(ctx, dynamicClient, restMapper, abstractObject, objectCache)
+		return client.GetTopController(abstractObject, objectCache)
 	}
 	return unstructuredObject, nil
 }
 
-func cacheAllObjectsOfKind(ctx context.Context, apiVersion, kind, namespace string, dynamicClient dynamic.Interface, restMapper meta.RESTMapper, objectCache map[string]unstructured.Unstructured) error {
+func (client Client) cacheAllObjectsOfKind(apiVersion, kind, namespace string, objectCache map[string]unstructured.Unstructured, mustNotHaveOwner bool) error {
```

Comment:
nit: I feel like `mustHaveOwner` would be less confusing to use

Reply:
Changed to `mustBeTopLevel`. Haven't had coffee yet but I think `mustHaveOwner` implies something different than `!mustNotHaveOwner`

Reply:
Much better.